### ISSUE TITLE
Changes in generated markdown file

### ISSUE THREAD

**Issue by spicy-sauce** [simonwalz/jsonschema2mk#5]
Hi,
Started to use this tool which is absolutely great, well done.

Here how I run it:
```
cat > ./docs/reference/config-reference.md <<EOF
          ---
          parent: Reference
          nav_order: 1
          ---

          EOF
          npx jsonschema2mk --schema ./schemas/config.schema.json --extension yaml-examples >> ./docs/reference/config-eference.md
```
A few issues:
1. As can be seen, as I use Github Pages' Jekyll just the docs theme, I had to add this front matter by hand. Would it be possible to have this in an additional argument?
2. Here's how the beginning of the generated file (with that front matter) looks like:
```
---
parent: Reference
nav_order: 1
---

<a name="root"></a>

#

Configuration File
```
The Configuration File header should be next to the #, like:
```
# Configuration File
```
3. All the generated `<a name ="<NAME>"></a>` should be removed. The internal links are OK
For example:
```
[**target**](#connectionstarget)
```
Will take you to:
```
### connections\.target:
```
without the need of this:
```
<a name="connectionstarget"></a>
```

Any chance to take care of those?
Thank you!

**Comment by spicy-sauce** [comment]
I'll open a PR for items 2 and 3, we can deal with 1 separately or leave it as is as you can still manipulate it yourself as in my example.

**Comment by spicy-sauce** [comment]
2 is not an issue really as I had no title in my jsonschema, just a description.

**Comment by spicy-sauce** [comment]
I'm closing this issue for now, it seems more related to how the just the docs builds the documentation based on the MD files rather an issue with this package.

**Comment by simonwalz** [comment]
Hi,

I am interested in your feedback and your suggestions.

No. 1
This can be achieved by using `--partials dir` with a customized main.md partial like the following:

```md
---
parent: Reference
nav_order: 1
---

{{> element schema path=""}}
```

I assume that the options like parent and nav_order are individual for every user. Is this right? I could think of this as a new extension but I am not sure about the options, that should be included here. What do you think?

No. 2
We should think of integration an alternative text, if no title is set. Thanks for the hint, that the field is left empty otherwise.

No. 3.
I've come across Markdown converters that don't insert `id` attributes or an `a` element. My solution was therefore to insert such an element. Does it bother you? What are the implications?

Kind regards
Simon

**Comment by spicy-sauce** [comment]
Hi,
Thanks for your response.
Answers:
1. Basically it's about adding a front matter to the generated MD file and since this is something individual per file, I'm not sure that using the `partials` is how this should be achieved unless you add such a customized `main.md` file per file. I think that having an option to plant such a front matter via the CLI itself is preferred, for example:
`npx jsonschema2mk  --schema config.schema.json --extension yaml-examples --front-matter parent: Reference\nnav_order: 1`
 
 2. Yes, although I'm not sure how common it is not to have the title set as I accidently did. Maybe if it had something like:
 `# [title not set]` instead it was more helpful, but I'm not sure if this is the expected output. You can consider adding a logger to the CLI so generated file won't be printed to the standard output but to the `--output-file` argument for example. So we will see a warning message like:
 [WARN] missing title in the root of your jsonschema
And finally:
[INFO] generated configutation-reference.md file successfully.

3. The first `<a name="root"></a>` did bother me as it appears above the header, so I had to do this:
`sed -i '/<a name="root"><\/a>/d' ./docs/reference/config-yaml-reference.md
`
The other ones do not bother the tool I use, although I wouldn't expect to have HTML markup there for this purpose. Maybe we should have a flag that determines the appearance of those anchors or not (in case they are needed somewhere... can you share more information about this?)

BTW, the `<br/>` tags are also redundant, you can just add an empty line to achieve the same goal. The only place you need explicit `<br/>` is in a table column.

Best,
Yossi

**Comment by simonwalz** [comment]
Hi Yossi

1. I added an extension: See [documentation](https://github.com/simonwalz/jsonschema2mk#internal-feature-extensions-option-extension), [code (js)](https://github.com/simonwalz/jsonschema2mk/blob/master/extensions/front-matter.js), [code (partial)](https://github.com/simonwalz/jsonschema2mk/tree/master/extensions/front-matter_partials)
2. Now the type is displayed if the title is not set. I think this produces usable output in all cases.
3. I removed the first `<a name="root"></a>`. It was not used. I will research if there is another solution for the other links.
4. (the BTW): Accouring to Wikipedia "two spaces at the end of a line produce a line break." I was not aware of that. I will look into making a change.

Any other improvment ideas?

Kind regards
Simon

**Comment by simonwalz** [comment]
I adapted the line breaks as well and published it as version 1.7.0.

**Comment by spicy-sauce** [comment]
Hi,
Just went over your changes and it looks fantastic, thank you so much for applying everything so fast and so well.
Will share any other improvements ideas in the future :-)

Thanks again,
Yossi
